### PR TITLE
Centralize authorization checks.

### PR DESCRIPTION
**Pull request by GunnarFarneback**
Do all authorization checks in one place.

### CODE REVIEW

**Review by ericphanson — APPROVED** [review]

File: Project.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name = "RegistryCI"
 uuid = "0c95cc5f-2f7e-43fe-82dd-79dbcba86b32"
 authors = ["Dilum Aluthge <dilum@aluthge.com>", "Fredrik Ekre <ekrefredrik@gmail.com>", "contributors"]
-version = "6.2.0"
+version = "6.3.0-DEV"
```
Comment (DilumAluthge) [inline Project.toml:4]: I generally prefer to release immediately after merging a PR. I.e. I only bump to a `-DEV` version if I'm planning on making multiple PRs in a row in a short period of time.

File: src/AutoMerge/types.jl [hunk 2]
```diff
@@ -90,6 +86,7 @@ end
 function GitHubAutoMergeData(;kwargs...)
     fields = fieldnames(GitHubAutoMergeData)
     @assert Set(keys(kwargs)) == Set(fields)
+    @assert kwargs[:authorization] ∈ (:normal, :jll)
```
Comment (DilumAluthge) [inline src/AutoMerge/types.jl:89]: I notice there are a couple other places where we now use `@assert`.

In a future version of Julia, assertions will be disabled by default. Actually, there is currently a PR open to JuliaLang/julia that will disable assertions by default.

So, in this package, I think it's best if we don't use `@assert`. Instead, we have the `always_assert` function.

If we like, we could replace the `always_assert` function with an `@always_assert` macro, if that is easier to use.

**Review by DilumAluthge — CHANGES_REQUESTED** [review]

File: src/AutoMerge/types.jl (hunk 2 above)
Comment (ericphanson) [reply src/AutoMerge/types.jl:89]: Good point, I forgot about that

**Comment by DilumAluthge** [comment]
bors merge

**Comment by DilumAluthge** [comment]
@GunnarFarneback See the test failure above.

**Comment by GunnarFarneback** [comment]
Yeah, things are not coming in quite the right order now.

**Comment by GunnarFarneback** [comment]
The easiest way to solve it is to do the new package/version check before the authorization check, if everybody is fine with that.

**Comment by DilumAluthge** [comment]
That's probably fine. We might eventually need to improve the `new_version_title_regex` and `new_package_title_regex` regexes, but we can do that in a different PR.

**Comment by DilumAluthge** [comment]
BTW you'll need to resolve the merge conflict.

**Comment by ericphanson** [comment]
bors merge

**Pull request merged**